From b25bdc159f6bbc4c46df2c52af0a16c349751019 Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Fri, 30 Sep 2016 13:18:59 -0400
Subject: [PATCH 57/58] crypto: keystone: make tx submission queue sharable

The both crypto sideband and inflow drivers use the same SA queue to
submit packets. We need to open it with KNAV_QUEUE_SHARED flag.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 drivers/crypto/keystone-sa.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/keystone-sa.c b/drivers/crypto/keystone-sa.c
index dfd4cf1..8c479ef 100644
--- a/drivers/crypto/keystone-sa.c
+++ b/drivers/crypto/keystone-sa.c
@@ -347,7 +347,8 @@ static int sa_setup_resources(struct keystone_crypto_data *dev_data)
 
 	snprintf(name, sizeof(name), "tx-subm-q-%s", dev_name(dev));
 	dev_data->tx_submit_q = knav_queue_open(name,
-						dev_data->tx_submit_qid, 0);
+						dev_data->tx_submit_qid,
+						KNAV_QUEUE_SHARED);
 	if (IS_ERR(dev_data->tx_submit_q)) {
 		ret = PTR_ERR(dev_data->tx_submit_q);
 		dev_err(dev, "Could not open \"%s\": %d\n", name, ret);
-- 
1.9.1

